sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        return response.json()
+
+    def get_enrichment_data_activation_job(self, data_activation_job_id):
+        """
+        Gets an enrichment data activation job with the given ID
+        :return: The API endpoint response as dictionary
+        """
+        response = self.get(DATA_ACTIVATION_ID_ENDPOINT.format(id=data_activation_job_id),
+                            log_msg_before=f'Getting enrichment data activation job {data_activation_job_id}',
+                            log_msg_after=f'Successfully got enrichment data activation job {data_activation_job_id}')
+        return response.json()[API_FIELD_VALUE]
+
+    def get_image_for_document(self, document_id, page_no: int):
+        """
+        Gets the image of a document page for the given document ID and page number
+        :param document_id: The ID of the document
+        :param page_no: The page number for which to get the image
+        :return: The image of the document page as bytearray
+        """
+        headers = {API_HEADER_ACCEPT: CONTENT_TYPE_PNG}
+        response = self.get(DOCUMENT_PAGE_ENDPOINT.format(document_id=document_id, page_number=page_no),
+                            headers=headers,
+                            log_msg_before=f'Getting image for page {page_no} of document with ID {document_id}',
+                            log_msg_after=f'Successfully got image for page {page_no} of document with ID {document_id}')
+        return response.content
+
+    def get_text_for_document(self, document_id, page_no: int):
+        """
+        Gets the text for a document page for the given document ID and page number
+        :param document_id: The ID of the document
+        :param page_no: The page number for which to get the text
+        :return: The text for the page as dictionary
+        """
+        response = self.get(DOCUMENT_PAGE_TEXT_ENDPOINT.format(document_id=document_id, page_number=page_no),
+                            log_msg_before=f'Getting text for page {page_no} of document with ID {document_id}',
+                            log_msg_after=f'Successfully got text for page {page_no} of document with ID {document_id}')
+        return response.json()[API_FIELD_VALUE]
+
+    def get_all_texts_for_document(self, document_id):
+        """
+        Gets the text for all pages of a document
+        :param document_id: The ID of the document
+        :return: A dictionary mapping the page numbers to the text
+        """
+        response = self.get(DOCUMENT_PAGES_TEXT_ENDPOINT.format(document_id=document_id),
+                            log_msg_before=f'Getting text for all pages of document with ID {document_id}',
+                            log_msg_after=f'Successfully got text fot all pages of document with ID {document_id}')
+        return response.json()[API_FIELD_RESULTS]
+
+    def get_request_for_document(self, document_id):
```

Comment:
Is this method required, isn't `get_results` sufficient for the end user?

Reply:
This method returns the contents of the request that was send when uploading the document. Those information are not included when getting the document via GET documents/{id}, which only includes the extracted results